lib/matplotlib/artist.py

```diff
@@ -1135,6 +1134,14 @@ def get_valid_values(self, attr):
         match = self._get_valid_values_regex.search(docstring)
         if match is not None:
             return re.sub("\n *", " ", match.group(1))
+
+        # Much faster than list(inspect.signature(func).parameters)[1],
+        # although barely relevant wrt. matplotlib's total import time.
+        param_name = func.__code__.co_varnames[1]
+        match = re.search("(?m)^ *{} : (.+)".format(param_name), docstring)
```

Comment:
I'm not quite sure, what we want the exact sematics to be.

What about `foo : int, optional, default: None`?

Do we want just `int` or `int, optional, default: None`?